cryptography-core/src/cdd_claim/cdd_claim_data.rs

```diff
@@ -37,22 +36,20 @@ impl CddClaimData {
 impl Encode for CddClaimData {
     #[inline]
     fn size_hint(&self) -> usize {
-        SCALAR_SIZE + SCALAR_SIZE
+        ScalarEncoder(&self.investor_did).size_hint()
+            + ScalarEncoder(&self.investor_unique_id).size_hint()
     }
 
     fn encode_to<W: Output>(&self, dest: &mut W) {
-        self.investor_did.as_bytes().encode_to(dest);
-        self.investor_unique_id.as_bytes().encode_to(dest);
+        ScalarEncoder(&self.investor_did).encode_to(dest);
+        ScalarEncoder(&self.investor_unique_id).encode_to(dest);
     }
 }
 
 impl Decode for CddClaimData {
     fn decode<I: Input>(input: &mut I) -> Result<Self, CodecError> {
-        let investor_did = <[u8; SCALAR_SIZE]>::decode(input)?;
-        let investor_did = Scalar::from_bits(investor_did);
-
-        let investor_unique_id = <[u8; SCALAR_SIZE]>::decode(input)?;
-        let investor_unique_id = Scalar::from_bits(investor_unique_id);
+        let investor_did = <ScalarDecoder>::decode(input)?.0;
+        let investor_unique_id = <ScalarDecoder>::decode(input)?.0;
```

Comment:
These could be abbreviated with a closure as well.

Reply:
See previous response.